WordPress/Classes/ViewRelated/Me/All Domains/Views/AllDomainsListViewController.swift

```diff
@@ -0,0 +1,149 @@
+
+    private lazy var state: ViewModel.State = viewModel.state
+
+    // MARK: - Observation
+
+    private var cancellable = Set<AnyCancellable>()
+
+    // MARK: - Init
+
+    init(viewModel: ViewModel = .init()) {
+        self.viewModel = viewModel
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - View Lifecycle
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.title = Strings.title
+        WPStyleGuide.configureColors(view: view, tableView: nil)
+        self.setupSubviews()
+        self.observeState()
+        self.viewModel.loadData()
+    }
+
+    private func setupSubviews() {
+        // Setup navigation item
+        self.navigationItem.rightBarButtonItem = .init(systemItem: .add)
+
+        // Setup search bar
+        let searchController = UISearchController(searchResultsController: nil)
+        searchController.delegate = self
+        searchController.searchBar.delegate = self
+        searchController.searchBar.placeholder = Strings.searchBar
+        self.navigationItem.searchController = searchController
+        self.navigationItem.hidesSearchBarWhenScrolling = false
+        self.extendedLayoutIncludesOpaqueBars = true
+        self.edgesForExtendedLayout = .top
+
+        // Setup tableView
+        self.tableView.translatesAutoresizingMaskIntoConstraints = false
+        self.tableView.dataSource = self
+        self.tableView.delegate = self
+        self.tableView.sectionHeaderHeight = .leastNormalMagnitude
+        self.tableView.sectionFooterHeight = Layout.interRowSpacing
+        self.tableView.contentInset.top = Layout.interRowSpacing
+        self.tableView.register(AllDomainsListTableViewCell.self, forCellReuseIdentifier: CellIdentifiers.myDomain)
+        self.tableView.register(AllDomainsListActivityIndicatorTableViewCell.self, forCellReuseIdentifier: CellIdentifiers.activityIndicator)
+        self.view.addSubview(tableView)
+        self.view.pinSubviewToAllEdges(tableView)
+    }
+
+    private func observeState() {
+        self.viewModel.$state.sink { [weak self] state in
+            guard let self else {
+                return
+            }
+            self.state = state
+            switch state {
+            case .normal, .loading:
+                self.tableView.reloadData()
+            case .error:
+                break
+            case .empty:
+                break
+            }
+        }.store(in: &cancellable)
+    }
+
+    // MARK: - Types
+
+    private enum Layout {
+        static let interRowSpacing = Length.Padding.double
+    }
+
+    private enum CellIdentifiers {
+        static let myDomain = String(describing: AllDomainsListTableViewCell.self)
+        static let activityIndicator = String(describing: AllDomainsListActivityIndicatorTableViewCell.self)
+    }
+
+    typealias ViewModel = AllDomainsListViewModel
```

Comment:
Same suggestion to move type aliases up for discoverability

Reply:
Resolved in 5dc702fc27c4064fa57ef047f39bbdaf44eda080